server/web_ui/src/views/identityverification.rs

```diff
@@ -0,0 +1,533 @@
+        let request = match state {
+            IdentifyUserState::Start => IdentifyUserRequest::Start,
+            _ => {
+                return IdentifyUserTransition::Error {
+                    msg: CORRUPT_STATE_ERROR.to_string(),
+                }
+            }
+        };
+        // `None` here is hard coded as when we start we obv don't know the other person identifier!
+        let response = match Self::make_request(request, None).await {
+            Ok(res) => res,
+            Err(s) => return IdentifyUserTransition::Error { msg: s.to_string() },
+        };
+        return IdentifyUserTransition::from(response);
+    }
+
+    async fn get_transition_from_id_display_and_submit(self) -> IdentifyUserTransition {
+        let request = match &self.state {
+            IdentifyUserState::IdDisplayAndSubmit { other_id_valid } => {
+                // in no case this function should have been called with an invalid id, but if that's the case then we do nothing
+                if *other_id_valid {
+                    IdentifyUserRequest::Start
+                } else {
+                    return IdentifyUserTransition::DoNothing;
+                }
+            }
+            _ => {
+                return IdentifyUserTransition::Error {
+                    msg: CORRUPT_STATE_ERROR.to_string(),
+                }
+            }
+        };
+        let response = match Self::make_request(request, self.other_id).await {
+            Ok(res) => res,
+            Err(s) => return IdentifyUserTransition::Error { msg: s.to_string() },
+        };
+        return IdentifyUserTransition::from(response);
+    }
+
+    async fn get_transition_from_submit_code(self) -> IdentifyUserTransition {
+        let request = match &self.state {
+            IdentifyUserState::SubmitCodeFirst {
+                other_totp,
+                totp_valid,
+            }
+            | IdentifyUserState::SubmitCodeSecond {
+                other_totp,
+                totp_valid,
+                // in no case this function should have been called with an invalid code, but if that's the case then we do nothing
+            } => {
+                if *totp_valid {
+                    IdentifyUserRequest::SubmitCode {
+                        other_totp: other_totp.to_string(),
+                    }
+                } else {
+                    return IdentifyUserTransition::DoNothing;
+                }
+            }
+            _ => {
+                return IdentifyUserTransition::Error {
+                    msg: CORRUPT_STATE_ERROR.to_string(),
+                }
+            }
+        };
+        let response = match Self::make_request(request, self.other_id).await {
+            Ok(res) => res,
+            Err(s) => return IdentifyUserTransition::Error { msg: s.to_string() },
+        };
+        return IdentifyUserTransition::from(response);
+    }
+
+    async fn make_request(
```

Comment:
I think there is a generic version of this in lib.rs (do_request).

Reply:
yep, I was using it inside the `make_request` function, but now I reckon that namewise it doesn't make much sense. I think I should rather call it something like "do_typed_request" to differentiate from the already existing do_request